### ISSUE TITLE
Runtime does not work on Manjaro

### ISSUE THREAD

**Issue by plata** [PhoenicisOrg/scripts#1056]
This probably affects also other systems therefore I'm not sure anymore if we can use the runtime like we anticipated.
In this particular case, any installation crashes with `Application tried to create a window, but no driver could be loaded.`. Trying to set the `DISPLAY` environment variable explicitly results in `load_builtin_dll failed to load .so lib for builtin L"winex11.drv": libbsd.so.0`.

**Comment by madoar** [comment]
Do you know the reason for this error? Does it occur because of a missing system library?

**Comment by ImperatorS79** [comment]
Could you verify if that lib is in the runtime ?

When doing my test I had no problem with libbsd. We can still add it as a dependency in the .deb (like libc6:i386) if it is required.

**Comment by plata** [comment]
Just to be clear: if I do not add the runtime to `LD_LIBRARY_PATH`, everything works. So I can't imagine that a system library is missing.

**Comment by madoar** [comment]
Do we need to add the engine libs to the `LD_LIBRARY_PATH`? Is there an environment where Phoenicis doesn't work if we don't do so?

**Comment by plata** [comment]
If we do not add the runtime to `LD_LIBRARY_PATH`, the whole point of having a runtime is defeated because it will not be used.

**Comment by madoar** [comment]
I don't understand why we have this problem at all. It seems to me like we're adding more libraries than we need to (because when removing the `_wineEnginesDirectory` libraries it is working). How can this lead to an error? Are some of the libraries corrupted?

In addition I don't understand why we have two locations for the libraries:
- `this._wineEnginesDirectory + "/runtime/libX/` and
- `this.getLocalDirectory(subCategory, version) + "/libX/`

I think it would be better to add all library dependencies to `this.getLocalDirectory(subCategory, version) + "/libX/`, especially because different wine versions could require different dependencies right?

**Comment by plata** [comment]
I'm not quite sure what exactly is causing the issues. It might be that a library replaces a system library which leads to errors. @ImperatorS79 would it be enough to use the runtime as fallback, i.e. list it at the end of `LD_LIBRARY_PATH`?
@madoar you experience the problem on Linux Mint, right? Normally this should be very close to Ubuntu. If this is not working already, I'm really concerned that using the runtime introduces more problems than it solves. @qparis what do you think?

**Comment by ImperatorS79** [comment]
If you put the runtime in `this.getLocalDirectory(subCategory, version) + "/libX/`, that means you have the runtime for each wine version, which is insane.

If changing the order inside `LD_LIBRARY_PATH` solve it, do it. 

Basically, you will have to have some libs inside the runtime since for some of them you need the exact right version (e.g. libavcodec,...). I was always in favour of using system libraries when they are working.

**Comment by plata** [comment]
I've just tried to put the runtime at the end of `LD_LIBRARY_PATH`. Still the same error.

**Comment by madoar** [comment]
Are there any news about this? I would like to know which linux distributions are currently affected by this? As far as I understand it there are currently at least two affected distributions:

- Linux Mint
- Manjaro

right? Are there other affected distributions? (What about Ubuntu?)

Is there anything we can do to fix this?

**Comment by ImperatorS79** [comment]
I cannot test on Ubuntu because the runtime is missing libfreetype (which it should not be since it is added to it -> https://github.com/PhoenicisOrg/phoenicis-winebuild/blob/master/builders/scripts/builder_linux_x86_runtime), which prevent me to launch basic apps like 7-zip.

**Comment by madoar** [comment]
Is this a problem in winebuild? If yes I think you should add an issue in winebuild targeting this.

For now can't you install `libfreetype` globally on your system (i.e. `apt install libfreetype`) to give this a test and eventually debug this? When the library is finally added to our wine packages you could uninstall it again.

**Comment by ImperatorS79** [comment]
I get 
```
0031:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0031:err:winediag:nodrv_CreateWindow Unknown error (127).
```
This is strange because I never got that with wine 4.0.

**Comment by madoar** [comment]
@ImperatorS79 this is the same error I get in https://github.com/PhoenicisOrg/scripts/issues/1059

**Comment by ImperatorS79** [comment]
The runtime is also missing libbsd, now it works.

**Comment by madoar** [comment]
So it should work after the runtimes are rebuilt with `winebuild`?

**Comment by ImperatorS79** [comment]
It works for me.

**Comment by madoar** [comment]
@plata I think this should be fixed now. Is the issue still occurring for you?

**Comment by plata** [comment]
Yes. Just tried with a fresh Notepad++ installation. Installation itself was working but I could not start Notepad++. After removing the runtime from the ld path, it worked.

**Comment by madoar** [comment]
Is it the same library or is it missing a different library this time?

**Comment by plata** [comment]
Like I said before: it cannot be that something is missing because if I do not use the runtime at all, it works.

**Comment by madoar** [comment]
I mean the error when you have the runtime added.

**Comment by madoar** [comment]
As far I understand @ImperatorS79 the reasons for this error and similar ones like #1098 is because of missing libraries in the runtime. So if we can complete the libraries it should work

**Comment by plata** [comment]
Ah ok. I get:
```
wine: cannot find L"C:\\windows\\system32\\winemenubuilder.exe"
000b:err:wineboot:ProcessRunKeys Error running cmd L"C:\\windows\\system32\\winemenubuilder.exe -a -r" (2)
0009:fixme:ver:GetCurrentPackageId (0x2e9610 (nil)): stub
0009:fixme:dc:DeleteDC not deleting busy DC 0xdd0049 refcount 2
```
But I don't think that's the issue really. Apart from that, the only thing happening is a Wine window "Unknown exception".

**Comment by ImperatorS79** [comment]
I get unknown error too on ubuntu 19.04. Removing the runtime is worse because I have not i386 .deb installed except libc6.

**Comment by plata** [comment]
The runtime does not work in Flatpak either.
